File: app/src/main/java/nikeno/Tenki/appwidget/weatherwidget/WeatherWidgetPrefs.java
package nikeno.Tenki.appwidget.weatherwidget;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.util.Log;

import androidx.annotation.NonNull;

import nikeno.Tenki.Utils;

public class WeatherWidgetPrefs {
    static final String TAG = "WidgetSettings";

    static final String NAME_URL = "name";

    @NonNull
    private static SharedPreferences getPrefs(@NonNull Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    @NonNull
    private static String prefKey(int id, @NonNull String name) {
        return "widget_" + id + name;
    }

    public static void addWidgetConfig(@NonNull Context context, int id, String url) {
        getPrefs(context).edit().putString(prefKey(id, NAME_URL), url).apply();
    }

    @NonNull
    public static WeatherWidgetConfig getWidgetConfig(@NonNull Context context, int id) {
        String url = Utils.httpsUrl(getPrefs(context).getString(prefKey(id, NAME_URL), ""));;
        return new WeatherWidgetConfig(url);
    }

    public static void deleteWidgetConfig(@NonNull Context context, @NonNull int[] ids) {
        SharedPreferences.Editor edit = getPrefs(context).edit();

        for (int id : ids) {
            Log.d(TAG, "deleteWidgetConfig id:" + id);
            edit.remove(prefKey(id, "url"));
        }
        edit.apply();
    }
}

File: app/src/main/java/nikeno/Tenki/appwidget/weatherwidget/WeatherWidgetPrefs.kt
package nikeno.Tenki.appwidget.weatherwidget

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.util.Log

object WeatherWidgetPrefs {
    const val TAG: String = "WidgetSettings"

    const val NAME_URL: String = "name"

    private fun getPrefs(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    private fun prefKey(id: Int, name: String): String {
        return "widget_$id$name"
    }

    fun addWidgetConfig(context: Context, id: Int, url: String?) {
        getPrefs(context).edit().putString(prefKey(id, NAME_URL), url).apply()
    }

    fun getWidgetConfig(context: Context, id: Int): WeatherWidgetConfig {
        val url = getPrefs(context).getString(prefKey(id, NAME_URL), "")
        return WeatherWidgetConfig(url)
    }

    fun deleteWidgetConfig(context: Context, ids: IntArray) {
        val edit = getPrefs(context).edit()

        for (id in ids) {
            Log.d(TAG, "deleteWidgetConfig id:$id")
            edit.remove(prefKey(id, "url"))
        }
        edit.apply()
    }
}
